Regression in 1.7.0.CR1: Maven build fails with "Failure to find org.gradle:gradle-tooling-api:jar:6.5"

**Describe the bug**
Switching from 1.6.1.Final to 1.7.0.CR1 in an _existing Maven project_ results in a build failure:
```
[INFO] Building middleware-shared-test-quarkus 0.1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for org.gradle:gradle-tooling-api:jar:6.5 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.455 s
[INFO] Finished at: 2020-08-01T01:06:46+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project middleware-shared-test-quarkus: Could not resolve dependencies for project com.somecompany.someproject.middleware:middleware-shared-test-quarkus:jar:0.1.0-SNAPSHOT: Failure to find org.gradle:gradle-tooling-api:jar:6.5 in https://artifacts.somecompany.de/artifactory/prj-someproject-build-release was cached in the local repository, resolution will not be reattempted until the update interval of someproject-release has elapsed or updates are forced -> [Help 1]
```
Yes, this is behind a corporate Artifactory server acting as a "proxy".

**Expected behavior**
I can just build/test my project with Maven as before, without adding any Gradle repos.

**Actual behavior**
Build fails until I add https://repo.gradle.org/gradle/libs-releases to the Maven repositories in pom.xml (or to my Artifactory virtual repo).

**To Reproduce**
Steps to reproduce the behavior:
1. n/a at this time

**Configuration**
n/a

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `MINGW64_NT-10.0-18363 XXX 3.0.7-338.x86_64 2019-11-21 23:07 UTC x86_64 Msys`
 - Output of `java -version`: `OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)`
 - GraalVM version (if different from Java):  n/a
 - Quarkus version or git rev: 1.7.0.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3

**Additional context**
After adding the repo, `mvn dependency:tree` shows:
```
[INFO] +- io.quarkus:quarkus-junit5:jar:1.7.0.CR1:compile
[INFO] |  +- io.quarkus:quarkus-bootstrap-core:jar:1.7.0.CR1:compile
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-app-model:jar:1.7.0.CR1:compile
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-maven-resolver:jar:1.7.0.CR1:compile
[...]
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-gradle-resolver:jar:1.7.0.CR1:compile
[INFO] |  |  |  \- org.gradle:gradle-tooling-api:jar:6.5:compile
```
It seems `gradle-tooling-api` 6.5 is not available via Maven central (repo1) or jcenter (these are the repos we have in our Artifactory virtual project repo).

The dependency was added here (transitively): https://github.com/quarkusio/quarkus/pull/10459/files#diff-e1d523d28b6eb842a1debab1b74c3297

I guess either the dependency has to be marked optional somehow, or the [Migration Guide](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.7) has to be updated to mention that you might need to add the Gradle repo.

--- a/independent-projects/bootstrap/core/pom.xml
+++ b/independent-projects/bootstrap/core/pom.xml
@@ -22,14 +22,14 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-bootstrap-maven-resolver</artifactId>
         </dependency>
-	<dependency>
+        <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-bootstrap-gradle-resolver</artifactId>
         </dependency>
         <dependency>
             <groupId>io.smallrye.common</groupId>
             <artifactId>smallrye-common-io</artifactId>
-	</dependency>
+        </dependency>
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter</artifactId>

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/IDELauncherImpl.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/IDELauncherImpl.java
@@ -12,12 +12,11 @@ import java.util.Map;
 
 /**
  * IDE entry point.
- * 
+ *
  * This is launched from the core/launcher module. To avoid any shading issues core/launcher unpacks all its dependencies
  * into the jar file, then uses a custom class loader load them.
- * 
+ *
  */
-@SuppressWarnings("unused")
 public class IDELauncherImpl {
 
     public static void launch(Path projectRoot, Map<String, Object> context) {

--- a/independent-projects/bootstrap/gradle-resolver/pom.xml
+++ b/independent-projects/bootstrap/gradle-resolver/pom.xml
@@ -32,4 +32,42 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <!--
+                 We are bundling the Gradle Tooling API classes
+                 to avoid the need to configure the gradle libs artifact repository
+                 in Quarkus application projects to be able to resolve it.
+            -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <minimizeJar>true</minimizeJar>
+                            <artifactSet>
+                                <includes>
+                                    <include>org.gradle:gradle-tooling-api</include>
+                                </includes>
+                            </artifactSet>
+                            <filters>
+                                <filter>
+                                    <artifact>*:*</artifact>
+                                    <excludes>
+                                        <exclude>META-INF/*</exclude>
+                                    </excludes>
+                                </filter>
+                            </filters>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
 </project>

